include/eredis.hrl

```diff
@@ -1,13 +1,14 @@
 %% Public types
 
+-type secret() :: fun(() -> iodata()).
 -type reconnect_sleep() :: no_reconnect | integer().
 -type registered_name() :: {local, atom()} | {global, term()} | {via, atom(), term()}.
 
 -type option() :: {host, string() | {local, string()}} |
                   {port, inet:port_number()} |
                   {database, integer()} |
-                  {username, iodata() | fun(() -> iodata()) | undefined} |
-                  {password, iodata() | fun(() -> iodata()) | undefined} |
+                  {username, iodata() | secret() | undefined} |
+                  {password, iodata() | secret() | undefined} |
```

Comment:
If `secret()` is used in the public API (as it is here in options) perhaps we should use clearer name such as `obfuscated_secret()` or just `obfuscated()`. (I prefer `obfuscated()` since it implies that it's something secret that is obfuscated.) We should probably also export the obfuscated secret type too so users can use it if the options are passed between functions before they're passed to eredis.